Introduce `ruff.printDebugInformation` command

snowsignal opened the pull request:
## Summary

Introduces a command to print debug information. It should look like this in the Output window:

<img width="632" alt="Screenshot 2024-06-10 at 12 21 07 PM" src="https://github.com/astral-sh/ruff-vscode/assets/19577865/671c38a0-00f6-4fd1-a4e6-824cc788b54f">

## Test Plan

See https://github.com/astral-sh/ruff/pull/11831 for testing information.

MichaReiser commented on src/extension.ts:
```diff
@@ -240,6 +240,19 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
       });
     }),
+    registerCommand(`${serverId}.printDebugInformation`, async () => {
+      if (!lsClient) {
```
Can we only register the command if this is the new LSP server because I suspect it isn't supported for `ruff-lsp`?

dhruvmanila replied:
Yeah, I agree. `ruff-lsp` doesn't have the `ruff.printDebugInformation` command in its capabilities so it'll throw an error.

dhruvmanila replied:
Confirmed that it'll raise an exception:
```
Failed to handle request 2 workspace/executeCommand ExecuteCommandParams(command='ruff.printDebugInformation', arguments=[{'uri': 'file:///Users/dhruv/playground/ruff/src/TCH002.py'}], work_done_token=None)
Traceback (most recent call last):
  File "/Users/dhruv/.local/state/pipx/venvs/ruff-lsp/lib/python3.12/site-packages/pygls/protocol/json_rpc.py", line 264, in _handle_request
    handler(params, msg_id)
  File "/Users/dhruv/.local/state/pipx/venvs/ruff-lsp/lib/python3.12/site-packages/pygls/protocol/lsp_meta.py", line 17, in decorator
    ret_val = base_func(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dhruv/.local/state/pipx/venvs/ruff-lsp/lib/python3.12/site-packages/pygls/protocol/language_server.py", line 320, in lsp_workspace__execute_command
    cmd_handler = self.fm.commands[params.command]
                  ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
KeyError: 'ruff.printDebugInformation'
```

snowsignal commented:
I've updated the command handler to ignore the command when we're using `ruff-lsp` (and I've added `(name server only)` to the command to make this more clear).

dhruvmanila commented on src/extension.ts:
```diff
@@ -191,6 +191,8 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
       });
     }),
     registerCommand(`${serverId}.executeFormat`, async () => {
+      // let configuration = getConfiguration(serverId) as ISettings;
+
```
Is this a leftover? ;)

snowsignal replied:
Good catch, I'll remove that
